src/main/java/hudson/plugins/sonar/AbstractMsBuildSQRunner.java

```diff
@@ -58,24 +59,33 @@ public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListen
 
   protected abstract void perform(Run<?, ?> run, FilePath workspace, Launcher launcher, TaskListener listener) throws InterruptedException, IOException;
 
-  protected String getExeName(MsBuildSQRunnerInstallation msBuildScanner, EnvVars env, Launcher launcher, TaskListener listener, FilePath workspace)
+  protected String getScannerPath(MsBuildSQRunnerInstallation msBuildScanner, EnvVars env, Launcher launcher, TaskListener listener, FilePath workspace)
     throws IOException, InterruptedException {
     MsBuildSQRunnerInstallation inst = BuilderUtils.getBuildTool(msBuildScanner, env, listener, workspace);
 
     String exe;
     if (inst != null) {
-      exe = inst.getExecutable(launcher);
+      exe = inst.getToolPath(launcher);
       if (exe == null) {
         throw new AbortException(Messages.MsBuildScanner_ExecutableNotFound(inst.getName()));
       }
     } else {
       listener.getLogger().println(Messages.MsBuildScanner_NoInstallation());
-      exe = EXE;
+      exe = MsBuildSQRunnerInstallation.getScannerName();
     }
 
     return exe;
   }
 
+  protected void addDotNetCommand(ArgumentListBuilder args) {
```

Comment:
Could allow later sprint to read path to `dotnet` from somewhere else but if you prefer I can duplicate the logic in path the `begin` and `end` classes.

Reply:
That's ok